modules/upgrade/file.go

```diff
@@ -0,0 +1,28 @@
+package upgrade
+
+import (
+	"os"
+	"path/filepath"
+)
+
+var ModeExecutable os.FileMode = 0111
```

Comment:
Indeed, the last 1 is probably not relevant. `0110` is enough I think, only test if the current user or group can execute ?

Reply:
Don't think that is 100% correct either, since it will mark a file which as `0010` permissions as executable, but if the file is owned by your user, it is actually not. Tested this a bit locally, and a file owned by my user can indeed not execute some other file I own if its marked as `0077`. 

This answer on stackexchange seems to give a pretty detailed explanation https://unix.stackexchange.com/a/411901